packages/invoices/src/InboundShipment/DetailView/modals/InboundLineEdit/TabTables.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { FC } from 'react';
+import {
+  DataTable,
+  useColumns,
+  TextInputCell,
+  getLineLabelColumn,
+  NumberInputCell,
+  CurrencyInputCell,
+} from '@openmsupply-client/common';
+import { InboundShipmentRow } from '../../../../types';
+
+export const BatchTable: FC<{ batches: InboundShipmentRow[] }> = ({
+  batches,
+}) => {
+  const columns = useColumns<InboundShipmentRow>([
+    getLineLabelColumn(),
+    ['batch', { Cell: TextInputCell, width: 200 }],
+    [
+      'numberOfPacks',
```

Comment:
Ah yes. I've been wondering how best to approach it. Either somehow get a validate fn to the cell or doing the validation after the cell triggers to callback or just creating specific cells when they have specific validation rules.

I am leaning toward specific cells because it's easier..